ompi/mca/vprotocol/pessimist/vprotocol_pessimist_wait.c

```diff
@@ -19,7 +20,7 @@ static int vprotocol_pessimist_request_no_free(ompi_request_t **req) {
 }
 
 #define PREPARE_REQUESTS_WITH_NO_FREE(count, requests) do { \
-    for(int i = 0; i < count; i++) \
+    for (typeof(count) i = 0; i < count; i++)     \
```

Comment:
This is ridiculous, clearly non-human code to keep a tool happy.

Reply:
I don't understand?  There is a bug that will cause GCC warnings in the original code.  In both usages, this will expand out to:

```
int mca_vprotocol_pessimist_test_any(size_t count, ompi_request_t ** requests,
...
for (int i = 0; i < count; i++)
```

And is comparing an int and a size_t, which is wrong.  We could change `i` to be a size_t instead of an int, and that would avoid the compiler warning, but this is more generic.  So which part are you complaining about?  This isn't an automated formatting tool gone amok, this is an actual compiler warning and, honestly, a potential bug for sufficiently large counts.

Reply:
You do realize we are talking about the length of an array of requests passed to this function via the MPI API right ? So, count will never be anything than a positive value that totally fits into an int.

Anyway, that's not I complained about. The use of typeof in the for loop is a novelty in the OMPI code, and make a tool and humans puzzled.

Reply:
If your objection is using typeof() instead of size_t, that's fine, say that.  I get what you're saying about argument lengths, but do you get the importance of reducing compiler warning counts?  Because this case, sure, it should always be safe.  But we've had honest bugs in our code where we ignored the same exact warning.

Reply:
I really don't get your point about suddenly getting on your high horses and lecturing on compiler warnings. We wasted enough time on this 5 chars change, typeof it is.

Reply:
😄 

Ok.

Note that this PR fixes a bunch of warnings that generate a lot of noise, but it also fixes some warnings that were genuine bugs.  That's why I created this PR.

I don't have a strong feeling about fixing this particular issue with `typeof()`, but it seemed like an elegant solution.

Regardless: @bosilca is right -- we've spent enough time on this one.  `typeof()` it is.  Resolved.